armi/reactor/blocks.py

```diff
@@ -2192,6 +2175,23 @@ def _rotatePins(self, rotNum, justCompute=False):
 
         return rotateIndexLookup
 
+    def _rotatePinParameters(self, rotNum: int):
+        nPins = self.getNumPins()
+        params = self._getParamsWhere(
+            lambda pd: pd.atLocation(ParamLocation.CHILDREN)
+        )
+        for param in params:
+            name = param.name
+            original = self.p[name]
+            if isinstance(original, (list, np.ndarray)):
+                pass
+            elif isinstance(original, (int, float)):
+                pass
+            elif original is None:
+                pass
+            else:
+                raise TypeError(f"{name=} :: {original=}")
```

Comment:
```suggestion
            if isinstance(original, (list, np.ndarray, int, float, None)):
                pass
            elif original is not None:
                raise TypeError(f"{name=} :: {original=}")
```

Reply:
Not a demand. Just see what you think.

Reply:
Sorry this is very much draft still, not ready to go. Yes that's a better way to do this. But I need to add the special logic inside the list/array check to do the rotation